src/main/java/io/vertx/ext/jdbc/impl/actions/JDBCSQLRowStream.java

```diff
@@ -169,10 +176,9 @@ private void readRow(Future<JsonArray> fut) {
             result.addNull();
           }
         }
-        fut.complete(result);
-      } else {
-        fut.complete();
+        accumulator.add(result);
       }
+      fut.complete();
```

Comment:
+ a comment that says it is a volatile store and that this store is done *after* the accumulator write